Mathlib/RingTheory/Adjoin/Field.lean

```diff
@@ -97,6 +97,21 @@ theorem IsIntegral.mem_range_algebraMap_of_minpoly_splits [Algebra K L] [IsScala
     x ∈ (algebraMap K L).range :=
   int.mem_range_algHom_of_minpoly_splits h (IsScalarTower.toAlgHom R K L)
 
+theorem minpoly_add_algebraMap_splits [Algebra K L] {x : L} (r : K)
+    (g : (minpoly K x).Splits (algebraMap K L)) :
+    (minpoly K (x + algebraMap K L r)).Splits (algebraMap K L) := by
+  simpa [minpoly.add_algebraMap] using g.comp_X_sub_C r
+
+theorem minpoly_sub_algebraMap_splits [Algebra K L] {x : L} (r : K)
+    (g : (minpoly K x).Splits (algebraMap K L)) :
+    (minpoly K (x - algebraMap K L r)).Splits (algebraMap K L) := by
+  simpa only [sub_eq_add_neg, map_neg] using minpoly_add_algebraMap_splits (-r) g
+
+theorem minpoly_algebraMap_add_splits [Algebra K L] {x : L} (r : K)
```

Comment:
If you have this variant, then I feel we should also have the variant for `sub`. Does that make sense?

Reply:
Yes, of course! However, adding a `sub` version is surprisingly harder than I thought, because one needs to add similar polynomial lemmas for `neg`, which is not in mathlib yet. This will be added in a separate PR #17369